src/framework/mlt_slices.c

```diff
@@ -0,0 +1,224 @@
+/**
+ * \file mlt_slices.c
+ * \brief sliced threading processing helper
+ * \see mlt_slices_s
+ *
+ * Copyright (C) 2007-2016 Meltytech, LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include "mlt_slices.h"
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <pthread.h>
+
+#define MAX_SLICES 32
+#define ENV_SLICES "MLT_SLICES_COUNT"
+
+typedef struct mlt_slices_s
+{
+	int f_exit;
+	int count;
+	int readys;
+	pthread_mutex_t cond_mutex;
+	pthread_cond_t cond_var_job;
+	pthread_cond_t cond_var_ready;
+	pthread_t threads[MAX_SLICES];
```

Comment:
It seems like we also need a thread pool. If there are multiple services using slice processing along with frame-threaded parallel processing in the consumer, then there can be many threads active simultaneously. We can say this initial version should not be used with consumer parallel processing, but have you thought about a thread pool yet?